trollmoves/server.py

```diff
@@ -622,6 +627,46 @@ def _parse_delete(conf, raw_conf):
         conf["delete"] = val
 
 
+def _create_config_sub_dicts(original):
+    # Take a copy so we can modify the values if necessary
+    res = dict(original.items())
+    for key in original.keys():
+        parts = key.split("__")
+        if len(parts) > 1:
+            _create_dicts(res, parts, original[key])
+            del res[key]
+    return res
+
+
+def _create_dicts(res, parts, val):
+    cur = res
+    for part in parts[:-1]:
+        if part not in cur:
+            cur[part] = {}
+        cur = cur[part]
+    cur[parts[-1]] = _check_bool(val)
+
+
+def _check_bool(val):
+    if val.lower() in ["0", "false"]:
+        return False
+    elif val.lower() in ["1", "true"]:
+        return True
+    return val
+
+
+def _form_connection_parameters_dict(original):
+    # Take a copy so we can modify the values if necessary
+    res = dict(original.items())
+    if "connection_parameters" not in res:
+        res["connection_parameters"] = {}
+    for key in original.keys():
+        if key in CONNECTION_CONFIG_ITEMS:
+            res["connection_parameters"][key] = original[key]
```

Comment:
should we have a deprecation warning here? to motivate people to move to the __ syntax?

Reply:
Hmm. Maybe :thinking: I'll see think about this and see how to document this in the first place.

Reply:
Warning added in https://github.com/pytroll/trollmoves/pull/190/commits/e158f3a1fc2fc7a15a6e3872cd0ca45589b90080